File: src/services/smartReply.ts
import { ChatMessage, WardrobeItem } from '../types';
import { lunaApi } from './api';

export async function buildSmartReply(
  text: string,
  allItems: WardrobeItem[],
  base: Partial<ChatMessage>,
  token: string
): Promise<ChatMessage> {
  const norm = text.toLowerCase().trim();

  // Instant client-side replies — no network needed
  if (norm.match(/^(hi|hello|hey|hii|heyy|sup|yo)\b/)) {
    return {
      ...base,
      text: `Hey! I'm Luna, your personal stylist. Ask me what to wear, how to style something, or anything fashion — I've got you.`,
      intent: 'chat',
    } as ChatMessage;
  }

  if (norm.includes('thank')) {
    return {
      ...base,
      text: `Anytime! What else are we styling?`,
      intent: 'chat',
    } as ChatMessage;
  }

  if (norm.includes('help') || norm.includes('what can you do')) {
    return {
      ...base,
      text: `Here's what you can ask me:\n\n• "What should I wear to college tomorrow?"\n• "Show me all my black tops"\n• "What am I missing for winter?"\n• "Why am I a minimalist?"\n• "How do I style wide-leg trousers?"\n• "What colors go with burgundy?"\n\nJust talk to me like you'd text a stylist friend.`,
      intent: 'chat',
    } as ChatMessage;
  }

  // Everything else → Luna's LLM backend
  try {
    const result = await lunaApi.chat(text, token, allItems);
    return {
      ...base,
      text: result.reply,
      intent: 'chat',
    } as ChatMessage;
  } catch (error) {
    console.error('Luna LLM call failed:', error);
    return {
      ...base,
      text: `I'm having trouble reaching my AI right now. Try again in a moment — or ask me about outfits, your wardrobe, or style gaps.`,
      intent: 'chat',
    } as ChatMessage;
  }
}

File: smart_reply.py
import random

FALLBACKS = [
    "I'm best at helping you find outfits, search your wardrobe, or explain your style. Try asking me something like 'what should I wear today?'",
    "Not sure I caught that! You can ask me things like 'show my black tops' or 'what am I missing for winter'.",
    "I'm a styling assistant — ask me about your wardrobe, outfits, or your personal aesthetic!",
    "Hmm, I didn't quite get that. Try asking me to suggest an outfit or search for something specific in your wardrobe.",
]

GREETINGS = {"hi", "hello", "hey", "hii", "heyy", "sup", "yo"}

def get_smart_reply(message: str) -> str:
    text = message.lower().strip().rstrip("!")
    if text in GREETINGS:
        return "Hey! Ask me what to wear, search your wardrobe, or find out what you're missing. What do you need?"
    if any(word in text for word in ["thank", "thanks", "ty"]):
        return "Anytime! Anything else you want to explore in your wardrobe?"
    if any(word in text for word in ["help", "what can you do", "how does this work"]):
        return (
            "Here's what I can do:\n"
            "• **Outfit suggestions** — 'What should I wear to college tomorrow?'\n"
            "• **Wardrobe search** — 'Show all my black tops'\n"
            "• **Gap analysis** — 'What am I missing for winter?'\n"
            "• **Style explanation** — 'Why am I a minimalist?'"
        )
    return random.choice(FALLBACKS)
